### DISCUSSION TITLE
Salesforce (SFDX) and cypress integration

### DISCUSSION THREAD

**Discussion by hvaandres** [cypress-io/cypress#16629] [General]
I'm currently working on an integration with SFDX and cypress, and I would like to see if someone has a good idea of what else I can do with the following error that I'm getting: 

 Unexpected end of JSON input

Because this error occurred during a before all hook we are skipping the remaining tests in the current suite: New Account Creation.

![Capture](https://user-images.githubusercontent.com/26829975/119195182-71cf8c80-ba41-11eb-9b60-929d2587311e.JPG)

Code:

```
import * as faker from "faker";
describe("New Account Creation", function() {
  let instanceUrl:String;
  before(function() {
    // runs once before all tests in the block
    cy.exec(
      'sfdx force:org:display --json | sed -E "s/[[:cntrl:]][[0-9]{1,3}m//g"', {failOnNonZeroExit: false}
    ).then(response => {
      let result = JSON.parse(response.stdout).result;
      let sessionId = result.accessToken;
      instanceUrl = result.instanceUrl;

      cy.request(`${instanceUrl}/secur/frontdoor.jsp?sid=${sessionId}`);
      
  })
})
  it("Create Account", function() {
    var companyName = faker.company.companyName();
    cy.visit(`${instanceUrl}/lightning/o/Account/list`);
      cy.get(".forceActionLink[title=New]")
        .should("be.visible")
        .click();
      cy.wait(3000);
      cy.get(".uiLabel")
        .contains("Account Name")
        .get("input[type=text]")
        .eq(1)
        .type(companyName);
      cy.get(".slds-button[title=Save]").click();
    });
  });

```

Questions:

Do you think the logic is wrong?

The link to my repo is the following: https://github.com/hvaandres/salesforce-sfdx-_cypress_integration

**Comment by tit** [comment]
Why don't you want to use jsforce instead of sfdx?
